tests/posix/fs/src/test_fs_dir.c

```diff
@@ -88,6 +88,30 @@ static int test_lsdir(const char *path)
 	return res;
 }
 
+static int test_rmdir(void)
+{
+	int res;
+
+	TC_PRINT("\nmkdir tests:\n");
+
+	res = mkdir(TEST_DIR, S_IRWXG);
+	if (res) {
+		TC_PRINT("Error creating dir[%d]\n", res);
+		return res;
+	}
+
+	TC_PRINT("\nrmdir tests:\n");
+
+	res = rmdir(TEST_DIR);
+	if (res) {
+		TC_PRINT("Error removing dir[%d]\n", res);
+		return res;
+	}
+
+	TC_PRINT("\nremoved dir: %s\n",  TEST_DIR);
+	return res;
+}
```

Comment:
@karthi012 - this sort of pattern should suffice for the more modern style of tests. Can you please make some adjustments this this test and similar ones to the `remove` test?

```suggestion
ZTEST(posix_fs_dir_test, test_rmdir)
{
	/* degenerate cases */
	// rmdir ops that are expected to fail in various predictable ways 

	/* happy path */
	zassert_ok(mkdir(TEST_DIR, S_IRWXG), "Error creating dir: %d", errno);
	zassert_ok(rmdir(TEST_DIR), "Error removing dir: %d\n", errno);
}
```